src/org/zaproxy/zap/extension/saml/SAMLUtils.java

```diff
@@ -61,24 +61,26 @@ public static String b64Encode(byte[] data) {
      * @throws SAMLException
      */
     public static String inflateMessage(byte[] data) throws SAMLException {
-        try {
-            Inflater inflater = new Inflater(true);
-            inflater.setInput(data);
-            byte[] xmlMessageBytes = new byte[MAX_INFLATED_SIZE];
-            int resultLength = inflater.inflate(xmlMessageBytes);
-
-            if (!inflater.finished()) {
-                throw new SAMLException("Out of space allocated for inflated data");
+       try{
+           byte[] out = data;
+           int length = data.length;
+            try {
+                Inflater inflater = new Inflater(true);
+                inflater.setInput(data);
+                byte[] xmlMessageBytes = new byte[MAX_INFLATED_SIZE];
+                int inflatedLength = inflater.inflate(xmlMessageBytes);
+                if (!inflater.finished()) {
```

Comment:
Is it really essentially instant? Is any sort of wait or synchronization necessary?

Reply:
it is synchronized inside `finished()`

Reply:
Great